app/controllers/event_calendar_controller.ts

```diff
@@ -0,0 +1,52 @@
+import BaseController, {
+  CreateHookContext,
+  DeleteHookContext,
+  HookContext,
+  PartialModel,
+} from "#controllers/base_controller";
+import { BadRequestException } from "#exceptions/http_exceptions";
+import CalendarEvent from "#models/calendar_event";
+
+export default class EventCalendarController extends BaseController<
+  typeof CalendarEvent
+> {
+  protected readonly queryRelations = [];
+  protected readonly crudRelations = [];
+  protected readonly model = CalendarEvent;
+
+  protected async storeHook(
+    ctx: CreateHookContext<typeof CalendarEvent>,
+  ): Promise<PartialModel<typeof CalendarEvent> | void | undefined> {
+    const event = ctx.request as CalendarEvent;
```

Comment:
incorrect type: this is not a `CalendarEvent` object, but a partial object containing only the columns of the model.
The correct typing of this variable would be `Partial<InstanceType<typeof CalendarEvent>>`, or `PartialModel<typeof CalendarEvent>`
It is safe to assume that non-autogenerated and non-optional fields will be present, but there isn't a good way to represent this with TS.